ramips: add support for HiWiFi HC5861B

dengqf6 opened the pull request:
HiWiFi "Gee Enjoy1200" HC5861B is a dual-band router based on MediaTek MT7628AN
https://www.hiwifi.com/enjoy-view

Specifications:
- MediaTek MT7628AN 580MHz
- 128 MB DDR2 RAM
- 16 MB SPI Flash
- 2.4G MT7628AN 802.11bgn 2T2R 300Mbps
- 5G MT7612EN 802.11ac 2T2R 867Mbps
- 5x 10/100 Mbps Ethernet

Flash instruction:
1. Get SSH access to the router
2. SSH to router with `ssh -p 1022 root@192.168.199.1`, The SSH password is the same as the webconfig one
3. Upload OpenWrt sysupgrade firmware into the router's `/tmp` folder with SCP
4. Run `mtd write /tmp/<filename> firmware`
5. reboot

Commit:
ramips: add support for HiWiFi HC5861B

HiWiFi "Gee Enjoy1200" HC5861B is a dual-band router based on MediaTek MT7628AN
https://www.hiwifi.com/enjoy-view

Specifications:
- MediaTek MT7628AN 580MHz
- 128 MB DDR2 RAM
- 16 MB SPI Flash
- 2.4G MT7628AN 802.11bgn 2T2R 300Mbps
- 5G MT7612EN 802.11ac 2T2R 867Mbps
- 5x 10/100 Mbps Ethernet

Flash instruction:
1. Get SSH access to the router
2. SSH to router with `ssh -p 1022 root@192.168.199.1`, The SSH password is the same as the webconfig one
3. Upload OpenWrt sysupgrade firmware into the router's `/tmp` folder with SCP
4. Run `mtd write /tmp/<filename> firmware`
5. reboot

Everything is working

Signed-off-by: Deng Qingfang <dengqf6@mail2.sysu.edu.cn>

mkresin commented on target/linux/ramips/base-files/lib/ramips.sh:
```diff
@@ -232,6 +232,9 @@ ramips_board_detect() {
 	*"HC5861")
 		name="hc5861"
 		;;
+	*"HC5861B")
```
Please drop this block and use the board compatible string instead as userspace boardname (`hiwifi,hc5861b`)

mkresin commented on target/linux/ramips/image/mt76x8.mk:
```diff
@@ -58,6 +58,13 @@ define Device/hc5661a
 endef
 TARGET_DEVICES += hc5661a
 
+define Device/hc5861b
```
please use `define Device/hiwifi_hc5861b`here

mkresin commented on target/linux/ramips/image/mt76x8.mk:
```diff
@@ -58,6 +58,13 @@ define Device/hc5661a
 
+define Device/hc5861b
+  DTS := HC5861B
+  IMAGE_SIZE := $(ralink_default_fw_size_16M)
```
ralink_default_fw_size_16M is 16121856 byte == 0xF60000. But your firmware partition has a size of 0xf70000 bytes. Please use `15808k` here.

mkresin commented on target/linux/ramips/image/mt76x8.mk:
```diff
@@ -58,6 +58,13 @@ define Device/hc5661a
+  IMAGE_SIZE := $(ralink_default_fw_size_16M)
+  DEVICE_TITLE := HiWiFi HC5861B
+endef
+TARGET_DEVICES += hc5861b
```
`TARGET_DEVICES += hiwifi_hc5861b` please

mkresin commented on target/linux/ramips/dts/HC5861B.dts:
```diff
@@ -0,0 +1,142 @@
+	m25p80@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		linux,modalias = "m25p80", "w25q128";
```
Please drop the linux,modalias. it isn't used since kernel 4.4.

mkresin reviewed (changes requested):
Please use ```git commit --amend``` to add changes to your commit. Use ```git push --force``` to overwrite your github repository/branch. The PR will afterwards update automatically. Please don't close this PR and open a new one.

dengqf6 commented:
Updated

mkresin commented:
And for sure not tested. The userspace boardname used is now wrong:

> Please drop this block and use the board compatible string instead as userspace boardname (hiwifi,hc5861b)

dengqf6 commented:
Oops, forgot to replace it
SEDed

mkresin commented:
Please provide a bootlog! It doesn't make any sense if you're doing the requested changes without testing the outcome.

dengqf6 commented:
Here it is
```
BusyBox v1.29.2 () built-in shell (ash)

  _______                     ________        __
 |       |.-----.-----.-----.|  |  |  |.----.|  |_
 |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
 |_______||   __|_____|__|__||________||__|  |____|
          |__| W I R E L E S S   F R E E D O M
 -----------------------------------------------------
 OpenWrt SNAPSHOT, r7845-22b8237ba1
 -----------------------------------------------------
root@OpenWrt:~# dmesg
[    0.000000] Linux version 4.14.62 (f@userver) (gcc version 7.3.0 (OpenWrt GCC 7.3.0 r7489-2601e34fad)) #0 Fri Aug 10 16:43:23 2018
[    0.000000] Board has DDR2
[    0.000000] Analog PMU set to hw control
[    0.000000] Digital PMU set to hw control
[    0.000000] SoC Type: MediaTek MT7628AN ver:1 eco:2
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019655 (MIPS 24KEc)
[    0.000000] MIPS: machine is HiWiFi HC5861B
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 08000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] On node 0 totalpages: 32768
[    0.000000] free_area_init_node: node 0, pgdat 804a0060, node_mem_map 81000040
[    0.000000]   Normal zone: 256 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 32768 pages, LIFO batch:7
[    0.000000] random: get_random_bytes called from start_kernel+0x8c/0x474 with crng_init=0
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32512
[    0.000000] Kernel command line: console=ttyS0,115200 rootfstype=squashfs,jffs2
[    0.000000] PID hash table entries: 512 (order: -1, 2048 bytes)
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Writing ErrCtl register=00061b70
[    0.000000] Readback ErrCtl register=00061b70
[    0.000000] Memory: 124704K/131072K available (3711K kernel code, 176K rwdata, 852K rodata, 180K init, 209K bss, 6368K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 256
[    0.000000] intc: using register map from devicetree
[    0.000000] CPU Clock: 580MHz
[    0.000000] timer_probe: no matching timers found
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 6590553264 ns
[    0.000012] sched_clock: 32 bits at 290MHz, resolution 3ns, wraps every 7405115902ns
[    0.007895] Calibrating delay loop... 385.84 BogoMIPS (lpj=1929216)
[    0.074142] pid_max: default: 32768 minimum: 301
[    0.079034] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.085683] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.098665] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.108602] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.114829] pinctrl core: initialized pinctrl subsystem
[    0.120634] NET: Registered protocol family 16
[    0.234085] mt7620-pci 10140000.pcie: Port 0 N_FTS = 1b105000
[    0.389658] PCI host bridge /pcie@10140000 ranges:
[    0.394456]  MEM 0x0000000020000000..0x000000002fffffff
[    0.399717]   IO 0x0000000010160000..0x000000001016ffff
[    0.422075] mt7621_gpio 10000600.gpio: registering 32 gpios
[    0.427986] mt7621_gpio 10000600.gpio: registering 32 gpios
[    0.433753] mt7621_gpio 10000600.gpio: registering 32 gpios
[    0.440195] PCI host bridge to bus 0000:00
[    0.444288] pci_bus 0000:00: root bus resource [mem 0x20000000-0x2fffffff]
[    0.451280] pci_bus 0000:00: root bus resource [io  0xffffffff]
[    0.457216] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
[    0.464090] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    0.472133] pci 0000:00:00.0: [14c3:0801] type 01 class 0x060400
[    0.472175] pci 0000:00:00.0: reg 0x10: [mem 0x00000000-0x7fffffff]
[    0.472191] pci 0000:00:00.0: reg 0x14: [mem 0x20200000-0x2020ffff]
[    0.472271] pci 0000:00:00.0: supports D1
[    0.472283] pci 0000:00:00.0: PME# supported from D0 D1 D3hot
[    0.472740] pci 0000:01:00.0: [14c3:7662] type 00 class 0x028000
[    0.472812] pci 0000:01:00.0: reg 0x10: [mem 0x00000000-0x000fffff 64bit]
[    0.472870] pci 0000:01:00.0: reg 0x30: [mem 0x00000000-0x0000ffff pref]
[    0.472973] pci 0000:01:00.0: PME# supported from D0 D3hot D3cold
[    0.473211] pci_bus 0000:01: busn_res: [bus 01-ff] end is updated to 01
[    0.473232] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 01
[    0.473279] pci 0000:00:00.0: BAR 0: no space for [mem size 0x80000000]
[    0.479963] pci 0000:00:00.0: BAR 0: failed to assign [mem size 0x80000000]
[    0.486955] pci 0000:00:00.0: BAR 8: assigned [mem 0x20000000-0x200fffff]
[    0.493830] pci 0000:00:00.0: BAR 9: assigned [mem 0x20100000-0x201fffff pref]
[    0.501120] pci 0000:00:00.0: BAR 1: assigned [mem 0x20200000-0x2020ffff]
[    0.508005] pci 0000:01:00.0: BAR 0: assigned [mem 0x20000000-0x200fffff 64bit]
[    0.515392] pci 0000:01:00.0: BAR 6: assigned [mem 0x20100000-0x2010ffff pref]
[    0.522691] pci 0000:00:00.0: PCI bridge to [bus 01]
[    0.527699] pci 0000:00:00.0:   bridge window [mem 0x20000000-0x200fffff]
[    0.534572] pci 0000:00:00.0:   bridge window [mem 0x20100000-0x201fffff pref]
[    0.546299] clocksource: Switched to clocksource MIPS
[    0.552522] NET: Registered protocol family 2
[    0.557910] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.564919] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
[    0.571400] TCP: Hash tables configured (established 1024 bind 1024)
[    0.577945] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.583808] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.590460] NET: Registered protocol family 1
[    0.594887] PCI: CLS 0 bytes, default 32
[    0.598286] Crashlog allocated RAM at address 0x3f00000
[    0.605083] workingset: timestamp_bits=30 max_order=15 bucket_order=0
[    0.617407] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.623243] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.643661] io scheduler noop registered
[    0.647630] io scheduler deadline registered (default)
[    0.653608] Serial: 8250/16550 driver, 3 ports, IRQ sharing disabled
[    0.661173] console [ttyS0] disabled
[    0.664773] 10000c00.uartlite: ttyS0 at MMIO 0x10000c00 (irq = 28, base_baud = 2500000) is a 16550A
[    0.673967] console [ttyS0] enabled
[    0.680964] bootconsole [early0] disabled
[    0.689587] cacheinfo: Failed to find cpu0 device node
[    0.694799] cacheinfo: Unable to detect cache hierarchy for CPU 0
[    0.701730] spi-mt7621 10000b00.spi: sys_freq: 193333333
[    0.722134] m25p80 spi0.0: w25q128 (16384 Kbytes)
[    0.726981] 7 fixed-partitions partitions found on MTD device spi0.0
[    0.733422] Creating 7 MTD partitions on "spi0.0":
[    0.738303] 0x000000000000-0x000000030000 : "u-boot"
[    0.744278] 0x000000030000-0x000000040000 : "hw_panic"
[    0.750403] 0x000000040000-0x000000050000 : "factory"
[    0.756415] 0x000000050000-0x000000fc0000 : "firmware"
[    0.836682] 2 uimage-fw partitions found on MTD device firmware
[    0.842711] 0x000000050000-0x0000001d6268 : "kernel"
[    0.848682] 0x0000001d6268-0x000000fc0000 : "rootfs"
[    0.854494] mtd: device 5 (rootfs) set to be root filesystem
[    0.861834] 1 squashfs-split partitions found on MTD device rootfs
[    0.868177] 0x0000009b0000-0x000000fc0000 : "rootfs_data"
[    0.874572] 0x000000fc0000-0x000000fe0000 : "oem"
[    0.880278] 0x000000fe0000-0x000000ff0000 : "bdinfo"
[    0.886125] 0x000000ff0000-0x000001000000 : "backup"
[    0.892774] libphy: Fixed MDIO Bus: probed
[    0.909824] rt3050-esw 10110000.esw: link changed 0x00
[    0.916764] mtk_soc_eth 10100000.ethernet eth0: mediatek frame engine at 0xb0100000, irq 5
[    0.926991] NET: Registered protocol family 10
[    0.935555] Segment Routing with IPv6
[    0.939474] NET: Registered protocol family 17
[    0.944040] bridge: filtering via arp/ip/ip6tables is no longer available by default. Update your scripts to load br_netfilter if you need this.
[    0.957196] 8021q: 802.1Q VLAN Support v1.8
[    0.975697] VFS: Mounted root (squashfs filesystem) readonly on device 31:5.
[    0.983895] Freeing unused kernel memory: 180K
[    0.988436] This architecture does not have kernel memory protection.
[    2.285644] init: Console is alive
[    2.289425] init: - watchdog -
[    2.465855] random: fast init done
[    4.454273] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    4.697606] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    4.707368] init: - preinit -
[    6.617008] rt3050-esw 10110000.esw: link changed 0x00
[    7.126502] random: procd: uninitialized urandom read (4 bytes read)
[   10.338093] rt3050-esw 10110000.esw: link changed 0x01
[   10.507096] jffs2: notice: (359) jffs2_build_xattr_subsystem: complete building xattr subsystem, 3 of xdatum (2 unchecked, 1 orphan) and 20 of xref (1 dead, 0 orphan) found.
[   10.524865] mount_root: switching to jffs2 overlay
[   10.615707] overlayfs: upper fs does not support tmpfile.
[   10.637299] urandom-seed: Seeding with /etc/urandom.seed
[   10.870956] procd: - early -
[   10.873985] procd: - watchdog -
[   11.508109] procd: - watchdog -
[   11.511592] procd: - ubus -
[   11.682795] random: ubusd: uninitialized urandom read (4 bytes read)
[   11.694335] random: ubusd: uninitialized urandom read (4 bytes read)
[   11.701369] random: ubusd: uninitialized urandom read (4 bytes read)
[   11.708915] procd: - init -
[   12.460799] kmodloader: loading kernel modules from /etc/modules.d/*
[   12.510327] Initializing XFRM netlink socket
[   12.517961] NET: Registered protocol family 15
[   12.525683] tun: Universal TUN/TAP device driver, 1.6
[   12.545874] ipip: IPv4 and MPLS over IPv4 tunneling driver
[   12.656391] l2tp_core: L2TP core driver, V2.0
[   12.662996] l2tp_netlink: L2TP netlink interface
[   12.670659] sit: IPv6, IPv4 and MPLS over IPv4 tunneling driver
[   12.680758] l2tp_ip: L2TP IP encapsulation support (L2TPv3)
[   12.688693] l2tp_ip6: L2TP IP encapsulation support for IPv6 (L2TPv3)
[   12.708052] ip6_tables: (C) 2000-2006 Netfilter Core Team
[   12.746485] nf_conntrack version 0.5.0 (2048 buckets, 8192 max)
[   12.954296] Netfilter messages via NETLINK v0.30.
[   12.963347] ip_set: protocol 6
[   13.104429] Bridge firewalling registered
[   13.120001] Loading modules backported from Linux version wt-2017-11-01-0-gfe248fc2c180
[   13.128197] Backport generated by backports.git v4.14-rc2-1-31-g86cf0e5d
[   13.187392] gre: GRE over IPv4 demultiplexor driver
[   13.212392] ip_tables: (C) 2000-2006 Netfilter Core Team
[   13.638011] xt_time: kernel timezone is -0000
[   13.807597] mt76_wmac 10300000.wmac: ASIC revision: 76280001
[   13.818278] mt76_wmac 10300000.wmac: Firmware Version: 20151201
[   13.824343] mt76_wmac 10300000.wmac: Build Time: 20151201183641
[   13.830421] mt76_wmac 10300000.wmac: Firmware already running...
[   13.836524] firmware init done
[   13.843460] ieee80211 phy0: Selected rate control algorithm 'minstrel_ht'
[   13.859756] mt76x2e 0000:01:00.0: card - bus=0x1, slot = 0x0 irq=4
[   13.866282] mt76x2e 0000:01:00.0: ASIC revision: 76120044
[   13.951870] mt76x2e 0000:01:00.0: ROM patch build: 20141115060606a
[   14.215024] mt76x2e 0000:01:00.0: Firmware Version: 0.0.00
[   14.220646] mt76x2e 0000:01:00.0: Build: 1
[   14.224828] mt76x2e 0000:01:00.0: Build Time: 201507311614____
[   14.248309] mt76x2e 0000:01:00.0: Firmware running!
[   14.258913] ieee80211 phy1: Selected rate control algorithm 'minstrel_ht'
[   14.353206] PPP generic driver version 2.4.2
[   14.360945] PPP MPPE Compression module registered
[   14.368164] NET: Registered protocol family 24
[   14.374736] PPTP driver version 0.8.5
[   14.383526] l2tp_ppp: PPPoL2TP kernel driver, V2.0
[   14.393177] kmodloader: done loading kernel modules from /etc/modules.d/*
[   16.752229] urandom_read: 5 callbacks suppressed
[   16.752238] random: jshn: uninitialized urandom read (4 bytes read)
[   21.223822] rt3050-esw 10110000.esw: link changed 0x00
[   25.007396] rt3050-esw 10110000.esw: link changed 0x01
[   29.651125] device eth0 entered promiscuous mode
[   29.668587] br-lan: port 1(eth0.1) entered blocking state
[   29.674180] br-lan: port 1(eth0.1) entered disabled state
[   29.680117] device eth0.1 entered promiscuous mode
[   29.762350] br-lan: port 1(eth0.1) entered blocking state
[   29.767907] br-lan: port 1(eth0.1) entered forwarding state
[   42.826890] IPv6: ADDRCONF(NETDEV_UP): wlan0: link is not ready
[   42.862864] br-lan: port 2(wlan0) entered blocking state
[   42.868342] br-lan: port 2(wlan0) entered disabled state
[   42.874266] device wlan0 entered promiscuous mode
[   43.847569] IPv6: ADDRCONF(NETDEV_UP): wlan1: link is not ready
[   43.900334] br-lan: port 3(wlan1) entered blocking state
[   43.905737] br-lan: port 3(wlan1) entered disabled state
[   43.911629] device wlan1 entered promiscuous mode
[   44.542468] IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes ready
[   44.549210] br-lan: port 2(wlan0) entered blocking state
[   44.554614] br-lan: port 2(wlan0) entered forwarding state
[   46.650795] IPv6: ADDRCONF(NETDEV_CHANGE): wlan1: link becomes ready
[   46.657536] br-lan: port 3(wlan1) entered blocking state
[   46.662931] br-lan: port 3(wlan1) entered forwarding state
[   60.831864] random: crng init done
root@OpenWrt:~#
```

mkresin commented:
merged, thanks!
